verification/curator-service/src/model/source.ts

```diff
@@ -15,17 +15,68 @@ const originSchema = new mongoose.Schema({
     },
 });
 
+interface Field {
+    name: string;
+    regexp: string;
+}
+
+const fieldSchema = new mongoose.Schema({
+    url: {
+        type: String,
+        required: 'Enter a dotted.field.name',
+    },
+    regexp: {
+        type: String,
+        required: 'Enter a field extraction regexp',
+    },
+});
+
+interface Parsing {
+    fields: Array<Field>;
+}
+
+const parsingSchema = new mongoose.Schema({
+    fields: {
+        type: [fieldSchema],
+    },
+});
+
+interface Automation {
+    name: string;
+    tag: string;
+    active: boolean;
+    scheduleExpression: string;
+    parsing: Parsing;
+}
+
+const automationSchema = new mongoose.Schema({
+    name: {
+        name: {
+            type: String,
+            required: 'Enter an automation rule name',
+        },
+        tag: String,
+        active: Boolean,
+        scheduleExpression: String,
```

Comment:
Something to consider down the line:

https://mongoosejs.com/docs/validation.html#custom-validators

Reply:
Yeah I saw that, I'm waiting to see how we want to optionalize/make mandatory various manual vs automatic configs before adding more validation.
I left a TODO for a regexp validator like "cron(" or "rate("